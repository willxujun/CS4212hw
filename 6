//Source: Pathier vectors.jlite
class Main {
  Void main(Int argc, String argv){
    /*
     Main function
     Performs some calculations
    */
    String currentChar; // Unused
    Point p1;
    Point p2;
    Int buffer;
    Vector v;

    v = new Vector();
    v.p1 = new Point();
    p2 = new Point();
    v.getPoint().p1.x = 90;
    v.p1.y = 89;
    p2.x = 43 + 78 * 9 - 7;
    readln(buffer);
    p2.y = buffer;


    println(v.norm() * 7 + 45 * 9);

    return false;
  }
}

class Point {
  Int x;
  Int y;

  String toString() {
    return "Point (x = " + this.x + ", y = " + this.y + ")";
  }
}

class Vector {
  Point p1;
  Point p2;

  Int norm() {
    Int res;
    res = (p1.x - p2.x) * (p1.x - p2.x) + (p2.y - p1.y) * (p2.y - p1.y);
    res = math.function.should_be_float("OK", true, "Yes", 90)
            .sqrt(res);
    return res + 67 - 67;
  }
}
