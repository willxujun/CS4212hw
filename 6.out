Warning: main method has non-Void return type.
===== CData3 ===== 
class Test

===== CMtd3 ===== 
Int %Test_0 [Test this]
[]
    Int t0;
    t0 = 0;
    return t0;


