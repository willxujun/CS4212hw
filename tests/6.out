===== CData3 ===== 
class FieldAccessMain
class FieldAccess
    Int a

===== CMtd3 ===== 
Void %FieldAccessMain_0 [FieldAccessMain this]
[Int a, Int b, Int temp, FieldAccess f, FieldAccess _t0, Int _t1, Int _t2, Int _t3, Int _t4, Int _t5]
    _t0 = new FieldAccess();
    f = _t0;
    _t1 = %FieldAccess_0 [f, 5];
    b = _t1;
    _t2 = f.a;
    _t2 = 7;
    _t3 = f.a;
    _t4 = _t3 + b;
    _t5 = _t4 * 2;
    temp = _t5;
    println temp;

Int %FieldAccess_0 [FieldAccess this, Int x]
[Int _t6]
    _t6 = 6 + x;
    return _t6;


